dash.el

```diff
@@ -1702,6 +1702,72 @@ is discarded."
         (setq i (1+ i))))
     (-flatten-n 1 (nreverse re))))
 
+(defun dash--match-kv-normalize-form (match-form)
+  "Normalize kv MATCH-FORM.
+
+In the kv matchers the expected format is the kv specifier
+
+- &keys
+- &plist
+- &alist
+- &hash
+
+followed by key-value pairs in a flat list (see `-let' for
+complete grammar), where all odd positions are keys and all even
+positions are match-forms by which the values are bound.
+
+A shorthand notation exists which allows the match-forms be
+optionally left out and derived from the key name in the
+following fashion:
+
+- a key :foo is converted into `foo' match-form,
+- a key 'bar is converted into `bar' match-form,
+- a key \"baz\" is converted into `baz' match-form.
+
+That is, the entire value under the key is bound to the derived
+variable without any further destructuring.
+
+This is possible only when the form following the key is not a
+valid match-form (i.e. a keyword, quoted symbol or a string).
```

Comment:
describe valid match forms instead of an incomplete example of invalid match forms.